.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,72 @@
+name: Continuous-Integration
+on: [push] # This workflow triggers anytime a commit is pushed

```

Comment:
Should we have it run on pull requests as well? Not sure if there's really a benefit to it in this case.

Reply:
Sometimes I'll commit some changes just to save my spot  or in case I need to revert. With the new workflow, I just won't be able to push these changes to origin, right?

Reply:
If you do push them and rebase them later by squashing or whatever it will re run the CI steps since they're new commits(since rebasing makes copies of the commits you had). So if you squash some commits that were previously failing(which ends up making them pass) and force push, it will run the CI again on those commits and update the check mark/X.

This is why the most recent commit in this branch is the only one with a check/X next to it.

Reply:
I don't think there will be any benefit from running on pull requests either, considering the only time re-running would change the output is when files are changed, and that's already covered by running on push.

Reply:
Yeah I agree. Only thing that would need to be run only on pull requests is Danger, but that'll be covered in a different PR.